### PR TITLE
Tick import error

### LINKED ISSUE #913: warn user if tick import errors
Sometimes tick import has an error (e.g. due to #631 ).  It is not obvious when this happens - I have been unsure whether I missed the success notification, it is still processing, or it failed.

Make it obvious.  Briefly say what happened (maybe just say which tick failed). Dev builds could give more details.

Related: https://github.com/OpenBeta/open-tacos/issues/891

**How important is this to you** (Please pick one)

- Important

### PR DESCRIPTION
**Pull request by clintonlunn**
fixes #913
---
name: Pull request
about: Create a pull request
title: ''
labels: ''
assignees: ''

---

## What type of PR is this?(check all applicable)
- [x] refactor
- [ ] feature
- [ ] bug fix
- [ ] documentation
- [ ] optimization
- [ ] other

### Related Issues

Issue #913 

### What this PR achieves

- Integrated `react-toastify` for improved error notifications.
- Swapped `headlessui` modal for `LeanAlert` for UI consistency.

### Screenshots, recordings

error messages and the new import mountain project modal
image
image

### Notes
I removed some unused `isButton` logic that doesn't seem to be used anywhere anymore. This really simplified the component.

### CODE REVIEW

**Comment by clintonlunn** [comment]
@vnugent I've been testing this and I believe it is pretty much feature complete. The only reason I have been holding it in draft is to try to get better test coverage, but I am still struggling with mocking some of these loading states and toast messages. That can probably be improved down the road though.

File: src/components/users/ImportFromMtnProj.tsx [hunk 1]
```diff
@@ -95,120 +125,68 @@ export function ImportFromMtnProj ({ isButton, username }: Props): JSX.Element {
...
+            )}
+
+            {showInput && (
+              <button
+                type='button'
+                onClick={getTicks}
+                className='btn btn-primary'
+              >
+                {loading ? <Spinner /> : 'Get my ticks!'}
+              </button>
```
Comment (vnugent) [inline src/components/users/ImportFromMtnProj.tsx]: I'd disable the button if loading is in progress

```html
 <button
   disabled={loading}
   ... 
```

**Review by vnugent — CHANGES_REQUESTED** [review]

**Comment by vnugent** [comment]
I was able to continue clicking on "Get my tick!" while the import is in progress (the spinner is showing), causing  multiple  imports.  Don't worry about unit testing for now.  As discussed the workflow will likely change. 

Screenshot 2023-08-27 at 10 49 39 AM

**Comment by clintonlunn** [comment]
Got the button disabled while loading, Other than that, I think we want to cancel the request if the user clicks cancel. Working on that now

**Pull request merged**